### PR TITLE
[MM-15536/MM-15832] Only show one "Connect to Jira" menu option.

### PR DESCRIPTION
**Pull request by jfrerich**
**Summary**
This PR addresses the issue when a Jira instance exists, but the user has not yet attached their Jira account to MM.  The Post Menu pulldown previously showed "Connect to Jira" for each action, where each available action is `Create Jira Issue` and `Attach Jira Issue`.  There is not a need to show this menu option twice. 

This ticket resolves the issue by only showing a single "Connect to Jira" menu option.

Note to reviewers:  A more complete solution would be to encapsulate the `create jira` and `attach jira` components with a parent component.  For simplicity and time, this solution uses the `create issue` post modal as the master to serve this function and removes the connecting feature from the `attach jira issue` modal.

**Ticket Link**
https://mattermost.atlassian.net/browse/MM-15536
https://mattermost.atlassian.net/browse/MM-15832

### CODE REVIEW

File: webapp/src/components/post_menu_actions/attach_comment_to_issue/attach_comment_to_issue.jsx [hunk 1]
```diff
@@ -56,16 +56,6 @@ export default class AttachCommentToIssuePostMenuAction extends PureComponent {
                     {this.getLocalizedTitle()}
                 </button>
             );
-        } else {
```
Comment (levb) [inline webapp/src/components/post_menu_actions/attach_comment_to_issue/attach_comment_to_issue.jsx]: 1/5 consolidate lines 43-48, smething like
```
    const conn = this.props.connected || {};
    if (this.props.isSystemMessage || !conn.connected) {
            return null;
    }

    let content;
    ...
```

Comment (jfrerich) [reply webapp/src/components/post_menu_actions/attach_comment_to_issue/attach_comment_to_issue.jsx]: I like this idea as it removes the extra if statement.  I think we will eventually refactor these post menu options components and end up doing the same for the create issue post menu modal.  Submitting change soon.

File: webapp/src/components/post_menu_actions/attach_comment_to_issue/attach_comment_to_issue.jsx [hunk 2]
```diff
@@ -40,33 +40,20 @@ export default class AttachCommentToIssuePostMenuAction extends PureComponent {
     }
 
     render() {
-        if (this.props.isSystemMessage) {
+        const conn = this.props.connected || {};
```
Comment (crspeller) [inline webapp/src/components/post_menu_actions/attach_comment_to_issue/attach_comment_to_issue.jsx]: Why is this an object? Looking at the reducer for `connected` it looks like we save a bool. But the mapStateToProps doesn't seem to use that, it seems to just grab any state. Can we fix this strange behavior?

Comment (jfrerich) [reply webapp/src/components/post_menu_actions/attach_comment_to_issue/attach_comment_to_issue.jsx]: Hey @crspeller.  I add an isConnected selector in the latest commit.  Please review to see if that fix is what you had in mind.  Thanks

**Comment by jfrerich** [comment]
@jasonblais FYI, the latest commit will add the jira icon to the "Connect to Jira" post menu item

**Pull request merged**